### PR TITLE
Fix if statement when check analytics's vendor setting exist

### PR DESCRIPTION
**Pull request by zhouyx**
This is such a stupid mistake : ( I really should have added the test along with last PR 😢 
Will need to patch into canary

[#10577 comment](https://github.com/ampproject/amphtml/pull/10577#issuecomment-318436454)

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 1]
```diff
@@ -457,6 +458,38 @@ describe('amp-analytics', function() {
     });
   });
 
+  it('should only throw user error when override vendor transport', function() {
+    const errorSpy = sandbox.spy();
+    sandbox.stub(log, 'user', () => {
+      return {
+        error: (tag, msg) => {errorSpy(msg);},
```
Comment (lannka) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: error: errorSpy,

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js [hunk 2]
```diff
@@ -457,6 +458,38 @@ describe('amp-analytics', function() {
...
+    };
+    const analytics1 = getAnalyticsTag({
+      'requests': {'foo': 'https://example.com/${bar}'},
+      'triggers': [{'on': 'visible', 'request': 'foo'}],
+    }, {'type': 'xyz'});
+    return analytics.layoutCallback().then(() => {
+      expect(errorSpy).to.be.calledWith('Inline or remote config should not ' +
+          'overwrite vendor transport settings');
+      errorSpy.reset();
+      return analytics1.layoutCallback().then(() => {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: split into 2 test cases

Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: done

File: extensions/amp-analytics/0.1/test/test-amp-analytics.js (hunk 1 above)
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/test-amp-analytics.js]: done

**Pull request merged**